### ISSUE TITLE
`capabilitiesOptions` defined in `.homeycompose/drivers/templates/` not taken into account

### ISSUE THREAD

**Issue by OlivierZal** [athombv/apps-sdk-issues#294]
It's not explicitly mentioned in the example of the documentation, but `capabilitiesOptions` in `.homeycompose/drivers/templates/{template_id}.json` are not taken into account – even if they are not overridden in `drivers/{driver_id}/driver.compose.json`.

In my opinion we should be able to define them also in the `templates`, otherwise we are forced to duplicate them at every `driver` level – what makes `templates` loose of their interest, since `templates` intend to allow some factorization.

**Comment by WeeJeWel** [comment]
I think that should work? All properties in a template are merged with the driver.compose.json.

**Comment by OlivierZal** [comment]
I made multiple tests, it does not work:
- capabilitiesOptions/titles in template are not taken into account
- when moving capabilitiesOptions/titles to the driver but leaving the corresponding capabilities in the remplace, they are taken into account

I’ll try to create a branch with code and corresponding screenshots.

**Comment by OlivierZal** [comment]
@WeeJeWel,

Could it happen that template capabilitiesOptions are merged into driver.compose.json at first – as you say –, and then **fully** replaced by driver capabilitiesOptions instead of having a merge of all the capabilitiesOptions keys from both sources?

**Comment by WeeJeWel** [comment]
Yes, they are not merged per-capability!

**Comment by OlivierZal** [comment]
Maybe I badly explained, because I think it's ok not to merge per-capability, I was rather talking about removing template capabilitiesOptions keys which are not at all defined in driver.compose.json – while merging.

I will rather give concrete examples.

**Comment by OlivierZal** [comment]
@WeeJeWel, here it is:

- `target_temperature` is defined at the [template](https://github.com/OlivierZal/com.melcloud/blob/3cb0ab0168f348fcb5877c1807b3b63bfb03732a/.homeycompose/drivers/templates/defaults.json#L12) level – and not (re)defined at the [driver](https://github.com/OlivierZal/com.melcloud/blob/main/drivers/melcloud_atw/driver.compose.json) level

- if I define `capabilitiesOptions/target_temperature/title` at the driver level, `title: "Template Driver"` is taken into account:

<img width="335" alt="Capture d’écran 2023-03-15 à 00 44 16" src="https://user-images.githubusercontent.com/88216225/225167403-c9341d6c-c1ae-4f50-9938-247e073897f1.png">

![IMG_2427](https://user-images.githubusercontent.com/88216225/225167463-b7aca766-4d75-4d97-94bb-62c1c2c99e28.PNG)

- if I move the **whole** `capabilitiesOptions/target_temperature` to the template, neither `title: "Template Template"` nor `min/max` are taken into account anymore (the default title `"Température cible"` is displayed):

<img width="372" alt="Capture d’écran 2023-03-15 à 00 46 57" src="https://user-images.githubusercontent.com/88216225/225167966-c48bbc79-d191-4ff8-9a1d-1888472fbacf.png">

![IMG_2428](https://user-images.githubusercontent.com/88216225/225167997-79e210ea-428f-4762-9007-6720e324c504.PNG)

Is that more clear with this concrete example?

**Comment by OlivierZal** [comment]
Did I convince you @WeeJeWel 😊?
Or do you need some additional data?

**Comment by WeeJeWel** [comment]
I'm sorry but I still don't completely understand.

Can you share:

* `/driver/<driverId>/driver.compose.json` with `$extends`
* /.homeycompose/drivers/templates/<templateId>.json` including `capabilitiesOptions.target_temperature`.
* The resulted `/app.json` after running `homey app build`.

**Comment by OlivierZal** [comment]
@WeeJeWel, I've created a branch and [PR](https://github.com/OlivierZal/com.melcloud/pull/113/files) showing the configuration I've made (data migration from `driver_compose.json` to `.homeycompose/drivers/template`).

I confirm that `.homeycompose/drivers/template` [does extend](https://github.com/OlivierZal/com.melcloud/blob/334269fcc39bc27c1a0bd0702ca1d57cb8831eb0/drivers/melcloud_atw/driver.compose.json#L2) `driver_compose.json`.

And the [`app.json`](https://github.com/OlivierZal/com.melcloud/pull/113/files#diff-96677aa35debfefd031d9d34d9c70369754ee3acb2d9a9d4090e98612efee6f5) generated by the `homey app install` confirms the bug: `template/defaults/capabilitiesOptions` **is not replicated into it** (it disappeared after moving it and making `homey app…`) – while other template data is well replicated.

**Comment by OlivierZal** [comment]
@WeeJeWel, did I raise my case? Or do you need some more info?

**Comment by WeeJeWel** [comment]
Yes, well received, but it's not our highest priority right now :-)

**Comment by OlivierZal** [comment]
@WeeJeWel, understood and agreed with the very low priority. Just wanted the bug to be acknowledged :-)

**Comment by OlivierZal** [comment]
@WeeJeWel, I've just discovered [HomeyCompose.js](https://github.com/athombv/node-homey/blob/master/lib/HomeyCompose.js): I don't see any mention of `capabilitiesOptions`, so maybe it can explain...

**Comment by OlivierZal** [comment]
December 2025: this is still an issue.

**Comment by WeeJeWel** [comment]
Solved. `homey@3.12.0` now merges `capabilitiesOptions` for each capability separately.
